docs/ModelConfig.md

```diff
@@ -0,0 +1,748 @@
+## Model Configuration File
+
+Typically, model training in rslearn involves fine-tuning remote sensing foundation
+models for prediction tasks encapsulated in an rslearn dataset. Typically, the dataset
+will contain one or more satellite image layers that contain the input that should be
+passed to the model; these could be individual images, or image time series. The
+dataset should also contain one or more raster or vector layers that act as labels, so
+that the model can be trained to make predictions corresponding to those layers.
+
+The model configuration file is a PyTorch Lightning-style YAML file that defines a
+model architecture, the tasks that the model should be trained for, and how the
+training should interface with the underlying rslearn dataset.
+
+The overall configuration looks like this:
+
+```yaml
+model:
+  class_path: rslearn.train.lightning_module.RslearnLightningModule
+  init_args:
+    model:
+      # ... Model architecture configuration.
+    # ... Learning rate, scheduler, and other options.
+data:
+  class_path: rslearn.train.data_module.RslearnDataModule
+  init_args:
+    path: # path to the rslearn dataset.
+    task:
+      # ... the task that defines how to train the model.
+    inputs:
+      # ... how layers in the dataset should be used
+    # other data related options
+trainer:
+  # Lightning trainer options and callbacks.
+```
+
+The YAML is parsed by jsonargparse, so each section directly configures a Python class
+or other type, and you can check those classes in the rslearn codebase for more
+details.
+
+Below, we dive into each of these components. This documentation is intended to
+supplement the examples in [Examples](Examples.md).
```

Comment:
Not an issue for this PR, but only one of the examples linked there actually works. And the actual model training part hasn't been written yet. For the future, I think it would be useful to have a couple of practical examples to show a full model config end-to-end, maybe one for segmentation and one for regression, so the reader can follow along with the examples as they read through each of these sections.